addons/account/models/account_move_line.py

```diff
@@ -916,14 +916,16 @@ def _compute_tax_key(self):
                     'partner_id': line.partner_id.id,
                     'move_id': line.move_id.id,
                 })
+            elif line.display_type == 'cogs':
+                line.tax_key = False
```

Comment:
I don't understand why you need this change?

Reply:
Because with only https://github.com/odoo/odoo/pull/112296/commits/9a70231219baf8c623c8703e1ebdca42004ba541#diff-78f3e1847de8ca0acf28d72a412947a549acdb08142d1dadf7646363d7972cb0R928
, in some cases, cogs amls end up in `existing_after` but not in `needed_after`, triggering a wrong delete